csp/mod.ts

```diff
@@ -1,46 +1,147 @@
+  "child-src"?: string;
+
+  /** Applies to XMLHttpRequest (AJAX), WebSocket, fetch(), <a ping> or EventSource */
+  "connect-src"?: string[];
+
+  /** Defines the default policy for fetching resources */
+  "default-src"?: string[];
+
+  /** Defines valid sources of font resources (loaded via @font-face) */
+  "font-src"?: string[];
+
+  /** Defines valid sources for loading frames */
+  "frame-src"?: string[];
+
+  /** Defines valid sources of images */
+  "img-src"?: string[];
+
+  /** Restricts the URLs that application manifests can be loaded */
+  "manifest-src"?: string[];
+
+  /** Defines valid sources of audio and video, eg HTML5 <audio>, <video> elements */
+  "media-src"?: string[];
+
+  /** Defines valid sources of plugins, eg <object>, <embed> or <applet> */
+  "object-src"?: string[];
+
+  /** Defines valid sources for request prefetch and prerendering */
+  "prefetch-src"?: string[];
+
+  /** Defines valid sources of JavaScript */
+  "script-src"?: string[];
+
+  /** Defines valid sources for JavaScript <script> elements, but not inline script event handlers */
+  "script-src-elem"?: string[];
+
+  /** Defines valid sources for JavaScript inline event handlers */
+  "script-src-attr"?: string[];
+
+  /** Defines valid sources of stylesheets or CSS */
+  "style-src"?: string[];
+
+  /** Defines valid sources for stylesheets <style> elements and <link> elements with rel="stylesheet" */
+  "style-src-elem"?: string[];
+
+  /** Defines valid sources for inline styles applied to individual DOM elements */
+  "style-src-attr"?: string[];
+
+  /** Restricts the URLs which may be loaded as a Worker, SharedWorker or ServiceWorker */
+  "worker-src"?: string[];
+
+  /** Defines a set of allowed URLs which can be used in the src attribute of a HTML base tag */
+  "base-uri"?: string[];
+
+  /** Defines valid MIME types for plugins invoked via <object> and <embed> */
+  "plugin-types"?: string[];
+
+  /** Enables a sandbox for the requested resource similar to the iframe sandbox attribute */
+  "sandbox"?: string[];
+
+  /** Defines valid sources that can be used as an HTML <form> action */
+  "form-action"?: string[];
+
+  /** Defines valid sources for embedding the resource using <frame> <iframe> <object> <embed> <applet> */
+  "frame-ancestors"?: string[];
+
+  /** Restricts the URLs that the document may navigate to by any means */
+  "navigate-to"?: string[];
+
+  /** Instructs the browser to POST a reports of policy failures to this URI */
+  "report-uri"?: string;
+
+  /** Defines a reporting group name defined by a Report-To HTTP response header */
+  "report-to"?: string;
+
+  /**
+   * Deprecated:
+   * Prevents loading any assets over HTTP when the page uses HTTPS
+   */
+  "block-all-mixed-content"?: string[];
+
+  /**
+   * Deprecated:
+   * Used to specify information in the Referer header for links away from a page
+   */
+  "referrer"?: [
```

Comment:
I think this type is wrong. "referrer" accept only one of these values, right? It should be a enum:
```
"referrer"?: "no-referrer" | "no-referrer-when-downgrade" | "origin" | "origin-when-cross-origin" | "unsafe-url";
```

Reply:
right.